File: CS-Examples/02_Data/FindStringAndNumber.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using Spire.Xls;
using System.Text;
using System.IO;

namespace FindStringAndNumber
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, System.EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\FindCellsSample.xlsx");

            //Get the first worksheet
            Worksheet sheet = workbook.Worksheets[0];

            //Find cells with the input string
            CellRange[] textRanges = sheet.FindAllString("E-iceblue", false, false);

            //Create a string builder
            StringBuilder builder = new StringBuilder();

            //Append the address of found cells in builder
            if (textRanges.Length != 0)
            {
                foreach (CellRange range in textRanges)
                {
                    string address = range.RangeAddress;
                    builder.AppendLine("The address of found text cell is: " + address);
                }
            }
            else
            {
                builder.AppendLine("No cells that contain the text");
            }

            //Find cells with the input integer or double
            CellRange[] numberRanges = sheet.FindAllNumber(100, true);

            //Append the address of found cells in builder
            if (numberRanges.Length != 0)
            {
                foreach (CellRange range in numberRanges)
                {
                    string address = range.RangeAddress;
                    builder.AppendLine("The address of found number cell is: " + address);
                }
            }
            else
            {
                builder.AppendLine("No cells that contain the number");
            }

            //Save to txt file
            string result = "FindStringAndNumber_out.txt";
            File.WriteAllText(result, builder.ToString());

            //Launch the file
            OutputViewer(result);
        }
        private void OutputViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

    }
}

File: VB-Examples/02_Data/FindStringAndNumber.vb
Imports System.Collections
Imports System.ComponentModel
Imports Spire.Xls
Imports System.Text
Imports System.IO

Namespace FindStringAndNumber
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Instantiate a new workbook object.
            Dim workbook As New Workbook()

            ' Load the Excel file from the specified path.
            workbook.LoadFromFile("..\..\..\..\..\..\Data\FindCellsSample.xlsx")

            ' Access the first worksheet in the workbook.
            Dim sheet As Worksheet = workbook.Worksheets(0)

            'Find the cells with the input string
            Dim textRanges() As CellRange = sheet.FindAllString("E-iceblue", False, False)

            'Create a string builder
            Dim builder As New StringBuilder()

            'Append the address of the found cells in string builder
            If textRanges.Length <> 0 Then
                For Each range As CellRange In textRanges
                    Dim address As String = range.RangeAddress
                    builder.AppendLine("The address of found text cell is: " & address)
                Next range
            Else
                builder.AppendLine("No cells that contain the text")
            End If

            ' Find the ells with the input integer or double data
            Dim numberRanges() As CellRange = sheet.FindAllNumber(100, True)

            'Append the address of the found cells in string builder
            If numberRanges.Length <> 0 Then
                For Each range As CellRange In numberRanges
                    Dim address As String = range.RangeAddress
                    builder.AppendLine("The address of found number cell is: " & address)
                Next range
            Else
                builder.AppendLine("No cells that contain the number")
            End If

            ' Write the contents of the string builder to a text file with the specified name.
            Dim result As String = "FindStringAndNumber_out.txt"
            File.WriteAllText(result, builder.ToString())
            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the file
            OutputViewer(result)
		End Sub
		Private Sub OutputViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
